src/Phpro/SoapClient/Middleware/Middleware.php

```
@@ -44,6 +45,10 @@ public function afterResponse(ResponseInterface $response): ResponseInterface
 
     public function onError(Exception $exception)
     {
+        if (!$exception instanceof Throwable) {
+            return;
```

Ignoring the exception doesnt seem acceptable to me. Maybe best to create a MiddlewareException which is throwable and which can contain an httplug exception? 
In that new exception, we can add logic to check if it's an actual exception or dislpay a standard error string?
This logic change is also something which should be covered with a unit test.

---

Problem is, that interface has no methods, so even getting a message from it will trigger PHPStan.
In reality all those interface implementations will be throwable for as far as i could see.

---

Something like this shouldn't throw an error in phpstan I guess.
````php
class MiddlewareException extends \Http\Client\Exception\TransferException
{
    public static function fromHttPlugException(\Http\Client\Exception $exception): self
    {
        $message = $exception instanceof \Throwable ? $exception->getMessage() : 'Something went wrong:'.get_class($exception);
        $previous = $exception instanceof \Throwable ? $exception : null;
        
        return new self(
            $message,
            0,
            $previous
        );
    }
}

````